### PR TITLE
pre-commit: add more checks

### PR DESCRIPTION
**Pull request by myakove**
"flake8-comprehensions"
"flake8-mutable"
"flake8-simplify"

### CODE REVIEW

**Review by sarahbx — APPROVED** [review]
+1 Love it ❤️

File: ocp_resources/utils.py [hunk 1]
```diff
@@ -204,12 +204,7 @@ def _is_exception_matched(exp, exception_messages):
         if not exception_messages:
             return True
 
-        for msg in exception_messages:
-            # Prevent match if provided with empty string
```
Comment (sarahbx) [inline ocp_resources/utils.py:208]: Can we please retain this comment though ?

Comment (myakove) [reply ocp_resources/utils.py:208]: ACK

File: ocp_resources/resource.py [hunk 2]
```diff
@@ -200,7 +199,7 @@ def _cmp(self, other):
             return 1
 
 
-class classproperty(object):  # noqa: N801
+class ClassProperty:  # noqa: N801
```
Comment (sarahbx) [inline ocp_resources/resource.py:202]: The `noqa` comment can be removed from this line now that this has been addressed.

**Review by sarahbx — APPROVED** [review]
lgtm 👍

**Pull request merged**